### PR TITLE
Add assertion messages

### PR DESCRIPTION
**Pull request by dapplion**
Add meaningful assert messages to all `assert()` statements.

Extends the assert util to include comparator methods (`equal`, `lte`, `lt`, `gte`, `gt`) so the error message contains the asserted values.

Assert errors currently printing 
```
AssertionError: Expect false == true
```

will print
```
AssertionError: Current slot less than this target epoch's start slot: 351 >= 354
```

### CODE REVIEW

File: packages/lodestar-beacon-state-transition/src/block/operations/attesterSlashing.ts [hunk 1]
```diff
@@ -45,5 +45,5 @@ export function processAttesterSlashing(
     }
   });
 
-  assert(slashedAny);
+  assert.true(slashedAny, "should slash any");
```
Comment (dapplion) [inline packages/lodestar-beacon-state-transition/src/block/operations/attesterSlashing.ts]: What would be an appropriate assertion message here? I don't believe this one is clear to end users

File: packages/lodestar-beacon-state-transition/src/util/blockRoot.ts [hunk 2]
```diff
@@ -25,8 +25,8 @@ import {computeStartSlotAtEpoch} from "./epoch";
  * Return the block root at a recent [[slot]].
  */
 export function getBlockRootAtSlot(config: IBeaconConfig, state: BeaconState, slot: Slot): Root {
-  assert(slot < state.slot);
-  assert(state.slot <= slot + config.params.SLOTS_PER_HISTORICAL_ROOT);
+  assert.lt(slot, state.slot, "cannot get future epochs");
+  assert.lte(state.slot, slot + config.params.SLOTS_PER_HISTORICAL_ROOT);
```
Comment (dapplion) [inline packages/lodestar-beacon-state-transition/src/util/blockRoot.ts]: Could you recommend an assert message here?

File: packages/lodestar-beacon-state-transition/src/block/operations/attesterSlashing.ts (hunk 1 above)
Comment (wemeetagain) [reply packages/lodestar-beacon-state-transition/src/block/operations/attesterSlashing.ts]: maybe "no slashable validators for attester slashing found"

File: packages/lodestar-beacon-state-transition/src/util/blockRoot.ts (hunk 2 above)
Comment (wemeetagain) [reply packages/lodestar-beacon-state-transition/src/util/blockRoot.ts]: maybe "cannot get block root from slot more than SLOTS_PER_HISTORICAL_ROOT in the past"

File: packages/lodestar-beacon-state-transition/src/util/blockRoot.ts [hunk 3]
```diff
@@ -25,8 +25,8 @@ import {computeStartSlotAtEpoch} from "./epoch";
  * Return the block root at a recent [[slot]].
  */
 export function getBlockRootAtSlot(config: IBeaconConfig, state: BeaconState, slot: Slot): Root {
-  assert(slot < state.slot);
-  assert(state.slot <= slot + config.params.SLOTS_PER_HISTORICAL_ROOT);
+  assert.lt(slot, state.slot, "cannot get future epochs");
```
Comment (wemeetagain) [inline packages/lodestar-beacon-state-transition/src/util/blockRoot.ts]: maybe "cannot get block root for slot in the future"

File: packages/lodestar-beacon-state-transition/src/util/duties.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ export function getCommitteeAssignment(
 ): CommitteeAssignment {
 
   const next2Epoch = getCurrentEpoch(config, state) + 2;
-  assert(epoch <= next2Epoch);
+  assert.lte(epoch, next2Epoch, "too future epoch");
```
Comment (wemeetagain) [inline packages/lodestar-beacon-state-transition/src/util/duties.ts]: maybe "cannot get committee assignment for epoch more than two ahead"

File: packages/lodestar/src/chain/forkChoice/arrayDag/lmdGhost.ts [hunk 5]
```diff
@@ -341,7 +342,7 @@ export class ArrayDagLMDGHOST implements ILMDGHOST {
   }
 
   public headStateRoot(): Uint8Array {
-    assert(Boolean(this.justified));
+    assert.true(Boolean(this.justified), "Justified checkpoint does not exist");
```
Comment (wemeetagain) [inline packages/lodestar/src/chain/forkChoice/arrayDag/lmdGhost.ts:345]: Maybe we should standardize between capitalized and non-capitalized errors. No preference for either, just consistency

**Review by wemeetagain — CHANGES_REQUESTED** [review]
Looks great, the assert upgrade is really nice

**Comment by wemeetagain** [comment]
linter error: https://github.com/ChainSafe/lodestar/pull/1065/checks?check_run_id=814292314#step:7:84
otherwise looks good

**Review by mpetrunic — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**